test/liquidityConversionRate.js

```diff
@@ -0,0 +1,719 @@
+    it("should test calculation of buy rate for non zero quantity.", async function () {
+        let expectedResult = priceForDeltaE(feePercent, r, pMin, deltaE, e0).mul(precision).valueOf()
+        let result =  await liqConvRatesInst.buyRate(eInFp, deltaEInFp)
+        assertAbsDiff(result, expectedResult, expectedDiffInPct)
+    });
+
+    it("should test calculation of deltaE.", async function () {
+        let expectedResult = calcDeltaE(r, pMin, deltaT, e0).abs().mul(formulaPrecision).valueOf()
+        let result =  await liqConvRatesInst.deltaEFunc(rInFp, pMinInFp, eInFp, deltaTInFp, formulaPrecision, formulaPrecisionBits);
+        assertAbsDiff(result, expectedResult, expectedDiffInPct)
+    });
+
+    it("should test calculation of sell rate for non zero quantity.", async function () {
+        let expectedResult = priceForDeltaT(feePercent, r, pMin, deltaT, e0).mul(precision).valueOf()
+        let deltaTAfterReducingFeeInFp = BigNumber(deltaTInFp).mul((100 - feePercent) / 100);
+        let result =  await liqConvRatesInst.sellRate(eInFp, deltaTInFp, deltaTAfterReducingFeeInFp);
+        result = result[0]
+        assertAbsDiff(result, expectedResult, expectedDiffInPct)
+    });
+
+    it("should test recording of imbalance.", async function () {
+        let buyAmountInTwei = BigNumber(10).pow(tokenDecimals).mul(deltaT)
+
+        let expectedValueBeforeReducingFee = buyAmountInTwei.mul(3) / ((100 - feePercent)/100) // TODO - this calc is a duplication, move to general place...
+        let expectedResult = BigNumber(((feePercent / 100) * expectedValueBeforeReducingFee).toString())
+
+        await liqConvRatesInst.recordImbalance(token.address, buyAmountInTwei, 3000, 3000, {from: reserveAddress})
+        await liqConvRatesInst.recordImbalance(token.address, buyAmountInTwei.mul(2), 3000, 3000, {from: reserveAddress})
+        let result = await liqConvRatesInst.collectedFeesInTwei()
+
+        assertAbsDiff(result, expectedResult, expectedDiffInPct)
+    });
+
+    it("should test resetting of imbalance.", async function () {
+        let beforeReset = await liqConvRatesInst.collectedFeesInTwei();
+        assert.notEqual(beforeReset, 0, "bad result");
+
+        await liqConvRatesInst.resetCollectedFees()
+        let result = await liqConvRatesInst.collectedFeesInTwei()
+        let expectedResult = 0
+        assert.equal(result, expectedResult, "bad result");
+    });
+
+    it("should test resetting of imbalance not by admin.", async function () {
+        try {
+            await liqConvRatesInst.resetCollectedFees({from:operator})
+            assert(false, "expected to throw error in line above.")
+        }
+        catch(e){
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+    });
+
+    it("should test getrate for buy=true and qtyInSrcWei = non_0.", async function () {
+        let expectedResult = priceForDeltaE(feePercent, r, pMin, deltaE, e0).mul(precision).valueOf()
```

Comment:
this qtyInSrcWei 
should be qtySrcInWei.

like any other place "in" is used.
unless the In is removed all togather.

Reply:
no, it is actually in src wei, meaning either eth wei or token wei.
(the variable can have different resolution depending on buy/sell)